src/Range.ts

```diff
@@ -0,0 +1,50 @@
+import * as _ from './lodashMini';
+export class Range {
+    /**
+     * 
+     * @param _limits inclusive limits. All numbers outside of it are nor in nor out of the range
+     * @param _range range inside of the limits. All ranges are inclusive.
+     */
+    constructor(private _limits: [number, number], private _range: Array<[number, number]>) {
+        // blank
+    }
+
+    in(val: number): boolean {
+        let index = _.sortedIndexBy(this._range, [val, val], item => item[1]);
+        if (index >= this._range.length) {
+            index = this._range.length - 1;
```

Comment:
You can return false here. The `val` is greater than the upper limit in the last range.